catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -511,7 +529,26 @@ private QueryRequest validateQueryRequest(QueryRequest queryRequest)
                     "Cannot perform query with null query, either passed in from endpoint, or as output from a PreQuery Plugin");
         }
 
-        return queryRequest;
+        int queryPageSize = queryRequest.getQuery()
+                .getPageSize();
+
+        if (queryPageSize >= 1 && queryPageSize <= MAX_PAGE_SIZE) {
+            return queryRequest;
+        }
```

Comment:
May want to update Javadoc at the top of this method. I will continue the hero since that is not a code change.